src/components/RestuarantMenu.js

```diff
@@ -1,24 +1,13 @@
-import {useState, useEffect} from 'react';
+
 import Shimmer from './shimmer';
 import { useParams } from 'react-router-dom';
-import { MENU_API_URL } from '../utils/constants';
+import useRestuarantMenu from '../utils/useRestuarantMenu';
 
 const RestuarantMenu = () => {
-    const [resInfo, setResInfo] = useState(null);
-
     const {resId} = useParams();
     console.log(resId)
 
-    useEffect(() => {
-        fetchMenu();
-    }, []);
-
-    const fetchMenu = async () => {
-        const data = await fetch(`${MENU_API_URL}${resId}`);
-        const json = await data.json();
-        console.log(json);
-        setResInfo(json.data);
-    }
+    const resInfo = useRestuarantMenu(resId)
```

Comment:
Implementing custom hook - `useRestuarantMenu` to abstract the fetching data from the component